subscriptions-manager/src/main/java/io/arlas/subscriptions/db/mongo/MongoDBFactoryConnection.java

```diff
@@ -49,11 +51,14 @@ public MongoClient getClient() {
         CodecRegistry defaultCodecRegistry = MongoClientSettings.getDefaultCodecRegistry();
         CodecRegistry pojoCodecRegistry = fromRegistries(defaultCodecRegistry, fromProviders(pojoCodecProvider));
 
-        final MongoClient client = MongoClients.create(
-                MongoClientSettings.builder()
-                        .codecRegistry(pojoCodecRegistry)
-                        .applyToClusterSettings(builder -> builder.hosts(getServers())).build()
-        );
+        MongoClientSettings.Builder mongoBuilder = MongoClientSettings.builder()
+                .codecRegistry(pojoCodecRegistry)
+                .applyToClusterSettings(builder -> builder.hosts(getServers()));
+        if (!StringUtils.isEmpty(mongoDBConfiguration.username) &&
+                !StringUtils.isEmpty(mongoDBConfiguration.password)) {
+            mongoBuilder = mongoBuilder.credential(MongoCredential.createCredential(mongoDBConfiguration.username, mongoDBConfiguration.authDatabase, mongoDBConfiguration.password.toCharArray()));
```

Comment:
Not sure you need to reassign mongoBuilder, it doesn't look immutable

Reply:
Indeed! That's fixed.